core-plugin/src/com/google/gct/idea/debugger/CloudDebuggerClient.java

```diff
@@ -105,15 +110,27 @@ public void statusChanged(boolean login) {
           HttpRequestInitializer initializer = new HttpRequestInitializer() {
             @Override
             public void initialize(HttpRequest httpRequest) throws IOException {
+              HttpHeaders headers = new HttpHeaders();
               httpRequest.setConnectTimeout(timeout);
               httpRequest.setReadTimeout(timeout);
+              httpRequest.setHeaders(headers);
               credential.initialize(httpRequest);
             }
           };
 
           HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
-          cloudDebuggerClient = new Builder(httpTransport, JSON_FACTORY, initializer).setRootUrl(ROOT_URL)
-            .setApplicationName(GoogleLoginUtils.getCurrentPlatformName()).build().debugger();
+          String userAgent = USER_AGENT;
+          try {
+            userAgent += GoogleLoginUtils.getCurrentPlatformName()
```

Comment:
Why? Can't we move the platform name stuff out of login?  Doesn't seem to
belong there anyway.  Also I'm not convinced it's doing much that's useful.


On Fri, Jan 8, 2016 at 11:06 AM, Elliotte Rusty Harold <
notifications@github.com> wrote:

> In core-plugin/src/com/google/gct/idea/debugger/CloudDebuggerClient.java
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/327#discussion_r49204147>
> :
>
> >                credential.initialize(httpRequest);
> >              }
> >            };
> >
> >            HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
> > -          cloudDebuggerClient = new Builder(httpTransport, JSON_FACTORY, initializer).setRootUrl(ROOT_URL)
> > -            .setApplicationName(GoogleLoginUtils.getCurrentPlatformName()).build().debugger();
> > +          String userAgent = USER_AGENT;
> > +          try {
> > +            userAgent += GoogleLoginUtils.getCurrentPlatformName()
>
> Probably don't want to move it to common-lib because that would make
> common-lib depend on the login package.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/327/files#r49204147>
> .
>